### DISCUSSION TITLE
Coupons aren't entries?

### DISCUSSION THREAD

**Discussion by sbrow** [duncanmcclean/simple-commerce#1084] [Help] [answered]
I need to associate Coupons with certain users, which I didn't think would be difficult, until I realized that coupons are not a collection? Furthermore, their blueprint is hard-coded into the `Coupon` class.

Could you explain the rationale behind this decision, since everything else (AFAIK) is a 'normal', extensible entry / collection object?

Thank you, I'm sorry to be such a pest. 

---

**Footnote:** I believe I can save the link to the coupon on the user side, but since I'd like to wire up the connection in the `CouponRedeemed` event, it made more sense to save it on the coupon side.

**Comment by duncanmcclean** [comment, marked as answer]
As you've discovered, coupons are their own concept and have a fixed set of fields. There's a couple of reasons for this:

* Collections & entries come with a lot of things that don't really make sense for coupons (eg. titles, slugs, statuses)
* There's only so much that can be customized in the Control Panel by addons, and I wanted to have more flexibility.

There's probably some other pros for doing it that I can't remember right now 😅 

> Footnote: I believe I can save the link to the coupon on the user side, but since I'd like to wire up the connection in the CouponRedeemed event, it made more sense to save it on the coupon side.

Yes, you'll have to wire it up on the user side since the fields are fixed on the coupon side.
